File: corpus/cs2/0345-bakugan/solutions/a-optimal/bakugan.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int r;
    while (cin >> r && r != 0) {
        vector<int> m(r);
        for (int i = 0; i < r; ++i) {
            cin >> m[i];
        }
        
        vector<int> l(r);
        for (int i = 0; i < r; ++i) {
            cin >> l[i];
        }
        
        int score_m = 0, score_l = 0;
        bool bonus_given = false;
        
        for (int i = 0; i < r; ++i) {
            score_m += m[i];
            score_l += l[i];
            
            if (!bonus_given && i >= 2) {
                bool m_consec = (m[i] == m[i-1] && m[i-1] == m[i-2]);
                bool l_consec = (l[i] == l[i-1] && l[i-1] == l[i-2]);
                
                if (m_consec && l_consec) {
                    bonus_given = true;
                } else if (m_consec) {
                    score_m += 30;
                    bonus_given = true;
                } else if (l_consec) {
                    score_l += 30;
                    bonus_given = true;
                }
            }
        }
        
        if (score_m > score_l) cout << "M\n";
        else if (score_l > score_m) cout << "L\n";
        else cout << "T\n";
    }
    
    return 0;
}

File: corpus/cs2/0345-bakugan/solutions/a-optimal/bakugan.py
import sys

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    idx = 0
    while idx < len(input_data):
        r = int(input_data[idx])
        idx += 1
        if r == 0:
            break
            
        m = [int(x) for x in input_data[idx:idx+r]]
        idx += r
        l = [int(x) for x in input_data[idx:idx+r]]
        idx += r
        
        score_m = 0
        score_l = 0
        bonus_given = False
        
        for i in range(r):
            score_m += m[i]
            score_l += l[i]
            
            if not bonus_given and i >= 2:
                m_consec = (m[i] == m[i-1] == m[i-2])
                l_consec = (l[i] == l[i-1] == l[i-2])
                
                if m_consec and l_consec:
                    bonus_given = True
                elif m_consec:
                    score_m += 30
                    bonus_given = True
                elif l_consec:
                    score_l += 30
                    bonus_given = True
                    
        if score_m > score_l:
            print("M")
        elif score_l > score_m:
            print("L")
        else:
            print("T")

if __name__ == '__main__':
    main()
